buffer/src/main/java/io/netty/buffer/Unpooled.java

```diff
@@ -579,10 +579,10 @@ public static ByteBuf copiedBuffer(ByteBuffer... buffers) {
     public static ByteBuf copiedBuffer(CharSequence string, Charset charset) {
         ObjectUtil.checkNotNull(string, "string");
         if (CharsetUtil.UTF_8.equals(charset)) {
```

Comment:
> I look the doc about autovectorisation. It is about take element from array by index. Here does not do that.

That doc is just one case of autovectorization (it can happen while writing too) and anyway we perform seq.charAt(i) that inlined could lead to bejust an array access (ideally). 

My comment is more re to check what happen on ASM level in different cases before assume that a simple change would perform better just because the JMH bench shows it. The JMH numbers are just numbers, I always suggest to understand what's happening under the hood to be sure what they mean.

Reply:
@franz1981 , That's fine.  I will do more analysis. Tks for your review.

Reply:
> Before I modified, 'ByteBufUtil.writeAscii' did not perform vector optimization

Help me to understand: the new version in this PR can trigger auto-vectorization here?
Or, are you gonna change this part I have reviewed to make it happen, instead?

Reply:
@franz1981 I mean the version before I commited dit not trigger vertorzation optimize? Am I wrong?

Reply:
> So my commit don't changed the behavior.

Yep, probably :)
But if we have some chance to trigger it, it's a nice perf boost ;)

Reply:
Annother questtion. Can invoke `ByteBuf copiedBuffer(CharSequence string, int offset, int length, Charset charset)` in method `ByteBuf copiedBuffer(CharSequence string, Charset charset)` to reduce duplicated code? Diff between the two methods are  the length of string equals 0 is checked or not.